DirectProgramming/DPC++FPGA/Tutorials/Features/experimental/latency_control/src/CMakeLists.txt

```diff
@@ -28,7 +28,12 @@ endif()
 set(EMULATOR_COMPILE_FLAGS "-Wall ${WIN_FLAG} -fintelfpga -DFPGA_EMULATOR")
 set(EMULATOR_LINK_FLAGS "-fintelfpga")
 set(HARDWARE_COMPILE_FLAGS "-Wall ${WIN_FLAG} -fintelfpga")
-set(HARDWARE_LINK_FLAGS "-fintelfpga -Xshardware -Xsboard=${FPGA_BOARD} ${USER_HARDWARE_FLAGS}")
+if(FPGA_BOARD STREQUAL "intel_a10gx_pac:pac_a10")
+    # hyper-optimized-handshaking does not apply to Intel Arria 10® FPGAs
+    set(HARDWARE_LINK_FLAGS "-fintelfpga -Xshardware -Xsboard=${FPGA_BOARD} ${USER_HARDWARE_FLAGS}")
+else()
+    set(HARDWARE_LINK_FLAGS "-fintelfpga -Xshardware -Xshyper-optimized-handshaking=off -Xsboard=${FPGA_BOARD} ${USER_HARDWARE_FLAGS}")
```

Comment:
Why to you need to set it off?
Should the user know that? Should it be reflected in the README?

Reply:
> Why to you need to set it off?

This is to allow the same latency constraint working on both A10 and S10. With hyper-optimization on, the latency between the controlled load and store will be much higher than it is now on S10, so the latency control on A10 cannot be applied to S10.

> Should the user know that? Should it be reflected in the README?

I can add a "Discussion" section at the end and say something like "Hyper-optimization is turned off to make the same set of latency control parameters work on both A10 and S10. You can turn it back on and try your own parameters on S10."

Reply:
You can just put that comment in the CMakeLists.txt file rather than "hyper-optimized-handshaking does not apply to Intel Arria 10® FPGAs" which is not the reason why you apply it on S10 :)

Reply:
Okay, I can do that.